Lib/test/test_typing.py

```diff
@@ -3767,6 +3767,15 @@ def test_pickle(self):
                 with self.assertRaises(pickle.PicklingError):
                     pickle.dumps(UserAge, proto)
 
+    def test_missing__name__(self):
+        code = ("import typing\n"
+                "NT = typing.NewType('NT', int)\n"
+                )
+        ns = {}
+        exec(code, ns)
+
+        self.assertIs(ns['NT'].__module__, None)
```

Comment:
I don't really understand this test and why we *want* the value to be `None` here. The main thing is that `exec(code, ns)` doesn't throw `KeyError` right?

Reply:
Yup, it's should not raise `KeyError` in case if `__name__` is missed.

Reply:
Yes please, in case someone else decides to change the default value in the future. Then this test won't scare them :).